fintech-examples/fintech-server/src/test/java/de/adorsys/opba/fintech/server/FinTechServerTests.java

```diff
@@ -0,0 +1,261 @@
+    private static final String POSTFIX = ".json";
+
+    private static final Gson gson = new Gson();
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Autowired
+    protected MockMvc mvc;
+
+    @Test
+    @SneakyThrows
+    public void loginPostOk() {
+        auth("peter", "1234");
+    }
+
+    /**
+     *
+     * @param username
+     * @param password
+     * @return XSRF Token
+     */
+    @SneakyThrows
+    private String auth(String username, String password) {
+        LoginBody loginBody = new LoginBody(username, password);
+        MvcResult result = this.mvc
+                .perform(MockMvcRequestBuilders.post(FIN_TECH_AUTH_URL)
+                        .header("X-Request-ID", UUID.randomUUID().toString())
+                        .content(gson.toJson(loginBody))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(MockMvcResultMatchers.header().exists("X-XSRF-TOKEN"))
+                .andReturn();
+        return result.getResponse().getHeader("X-XSRF-TOKEN");
+    }
+
+    @Test
+    @SneakyThrows
+    public void loginPostUnAuthorized() {
+        final LoginBody loginBody = new LoginBody("peter", "12345");
+        this.mvc
+                .perform(MockMvcRequestBuilders.post(FIN_TECH_AUTH_URL)
+                        .header("X-Request-ID", UUID.randomUUID().toString())
+                        .content(gson.toJson(loginBody))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(MockMvcResultMatchers.status().isUnauthorized())
+                .andExpect(MockMvcResultMatchers.header().doesNotExist("X-XSRF-TOKEN"));
+    }
+
+    @Mock
+    TppBankSearchApi mockedTppBankSearchApi;
+
+    @Autowired
+    BankSearchService bankSearchService;
+
+    @Test
+    @SneakyThrows
+    public void bankSearchAuthorized() {
+        final String keyword = "affe";
+        final Integer start = 1;
+        final Integer max = 2;
+
+        bankSearchService.setTppBankSearchApi(mockedTppBankSearchApi);
+
+        when(mockedTppBankSearchApi.bankSearchGET(any(), any(), eq(keyword), eq(start), eq(max)))
+                .thenReturn(gson.fromJson(readFile(getFilenameBankSearch(keyword, start, max)), BankSearchResponse.class));
+
+        LoginBody loginBody = new LoginBody("peter", "1234");
+        String xsrfToken = auth(loginBody.username, loginBody.password);
+        bankSearch(keyword, start, max, xsrfToken);
+    }
+
+    /**
+     *
+     * @param keyword
+     * @param start
+     * @param max
+     * @param xsrfToken
+     * @return first BankUUID of found list
+     */
+    @SneakyThrows
+    private String bankSearch(String keyword, Integer start, Integer max, String xsrfToken) {
```

Comment:
You can return MvcResult without status expectation, so this code can be reused at bankSearchUnAuthorized

Reply:
done